docs/rules/no-shadow.md

```diff
@@ -65,34 +65,32 @@ function foo() {
 }
 ```
 
-### `hoist`
+### hoist
 
-The option has three settings:
+The `hoist` option has three settings:
 
-* `all` - reports all shadowing before the outer variables/functions are defined.
 * `functions` (by default) - reports shadowing before the outer functions are defined.
+* `all` - reports all shadowing before the outer variables/functions are defined.
```

Comment:
I don't want to speak for @pedrottimark, but I believe this is so that the default is first, which makes sense in my mind.